Mathlib/NumberTheory/NumberField/Embeddings.lean

```diff
@@ -1148,23 +1148,83 @@ lemma isReal_infinitePlace : InfinitePlace.IsReal (infinitePlace) :=
 
 end Rat
 
+namespace NumberField
+
+open InfinitePlace Module
+
+section TotallyRealField
+
 /-
 
 ## Totally real number fields
 
 -/
 
-namespace NumberField
-
 /-- A number field `K` is totally real if all of its infinite places
 are real. In other words, the image of every ring homomorphism `K → ℂ`
 is a subset of `ℝ`. -/
-class IsTotallyReal (K : Type*) [Field K] [NumberField K] where
+@[mk_iff] class IsTotallyReal (K : Type*) [Field K] [NumberField K] where
   isReal : ∀ v : InfinitePlace K, v.IsReal
 
+variable {K : Type*} [Field K] [NumberField K]
+
+theorem nrComplexPlaces_eq_zero_iff :
+    nrComplexPlaces K = 0 ↔ IsTotallyReal K := by
+  classical
+  simp [Fintype.card_eq_zero_iff, isEmpty_subtype, isTotallyReal_iff]
+
```

Comment:
Can you add a theorem saying that a subfield of a totally real number field is again totally real? It could even be an instance, actually.

Reply:
Well, it depends. If it is a `Subalgebra \Q K` then it can pick it, right? I mean, how do you want to state "subfield" otherwise?

Reply:
Usually, I just say `Algebra F K`... I don't use `Subfield` that much

Reply:
Hum, as a matter of fact, there is no instance saying that the `Subfield` of a number field is a number field

Reply:
I am adding both instances

Reply:
There is only [NumberField.of_intermediateField](https://leanprover-community.github.io/mathlib4_docs/Mathlib/NumberTheory/NumberField/Basic.html#NumberField.of_intermediateField) for the time being....